File: JAVA/SchorrKeyPair.java
import java.math.BigInteger;

/**
 * Lớp đại diện cho cặp khóa công khai/bí mật của Schnorr
 */
public class SchorrKeyPair {
    private final BigInteger privateKey; // Khóa bí mật (x)
    private final BigInteger publicKey; // Khóa công khai (y = g^x mod p)
    private final SchorrSignatureParams params; // Tham số hệ thống

    public SchorrKeyPair(BigInteger privateKey, BigInteger publicKey, SchorrSignatureParams params) {
        this.privateKey = privateKey;
        this.publicKey = publicKey;
        this.params = params;
    }

    // Getters
    public BigInteger getPrivateKey() {
        return privateKey;
    }

    public BigInteger getPublicKey() {
        return publicKey;
    }

    public SchorrSignatureParams getParams() {
        return params;
    }

    @Override
    public String toString() {
        return "SchorrKeyPair{" +
                "privateKey=" + privateKey.toString(16).substring(0, Math.min(32, privateKey.toString(16).length()))
                + "..." +
                ", publicKey=" + publicKey.toString(16).substring(0, Math.min(32, publicKey.toString(16).length()))
                + "..." +
                '}';
    }
}

File: C#/SchorrKeyPair.cs
using System.Numerics;

namespace Schnorr
{
    public class SchorrKeyPair
    {
        public BigInteger PrivateKey { get; }
        public BigInteger PublicKey { get; }
        public SchorrSignatureParams Params { get; }

        public SchorrKeyPair(
            BigInteger privateKey,
            BigInteger publicKey,
            SchorrSignatureParams parameters
        )
        {
            PrivateKey = privateKey;
            PublicKey = publicKey;
            Params = parameters;
        }

        public override string ToString()
        {
            string sx = PrivateKey.ToString("X");
            string sy = PublicKey.ToString("X");
            return $"SchorrKeyPair{{ x={sx.Substring(0, System.Math.Min(32, sx.Length))}..., y={sy.Substring(0, System.Math.Min(32, sy.Length))}... }}";
        }
    }
}
